src/dataflow-types/src/plan/reduce.rs

```diff
@@ -164,6 +163,35 @@ pub enum ReducePlan {
     Collation(CollationPlan),
 }
 
+impl Arbitrary for ReducePlan {
```

Comment:
What is `Arbitrary` and can we get an explainer here?

Reply:
The `Arbitrary` trait should document itself (see example here:
https://dev.materialize.com/api/rust-private/mz_expr/scalar/func/enum.BinaryFunc.html#impl-Arbitrary), but basically it generates arbitrary instances of the type in order to test some property of it. We have been using it in the protobuf support work to ensure that an arbitrary compute command can be translated to and back from protobuf without loss of information.

I have added a comment though, on this particular implementation for `ReducePlan`.